mvcc/backend/backend.go

```diff
@@ -330,10 +337,20 @@ func (b *backend) run() {
 	}
 }
 
+// createConcurrentReadTxs creates and sends read transactions
+// to the channels of all blocked CommittedReadTx() callers.
 func (b *backend) createConcurrentReadTxs() {
 	// do not allow too many concurrent read txs.
 	// TODO: improve this by having a global pending counter?
-	for i := 0; i < 100; i++ {
+	// TODO(jpbetz): This has the potential to backlog indefinitely under heavly load.
+	// If we're going to impose a limit here. We might want to do more to ensure we're
+	// managing context deadlines and cancelations also.
+
+	// Since bbolt already keeps track of how many concurrent open read txns there are,
+	// we can use its data to enforce our limit.
+	concurrentReadTxns := b.db.Stats().OpenTxN
+
+	for i := 0; i < (maxConcurrentReadTxns - concurrentReadTxns); i++ {
```

Comment:
We probably need to run some simple benchmarks to find out what is the typical value for concurrentReadTxns.

Reply:
It's 1 + number of committed reads. Double checked it with a benchmark. Adding a gauge metric to this PR for the `concurrentReadTxns` value.

Reply:
But we will need to determine what a reasonable limit might be. I'm not 100% convinced that # of concurrent requests is sufficient, got an idea for another approach where we weight requests by their response count.